pkg/serverless/trace/propagation/carriers.go

```diff
@@ -286,9 +290,61 @@ func extractTraceContextFromStepFunctionContext(event events.StepFunctionPayload
 	return tc, nil
 }
 
-// stringToDdSpanID hashes the Execution ARN, state name, and state entered time to generate a 64-bit span ID
-func stringToDdSpanID(execArn string, stateName string, stateEnteredTime string) uint64 {
-	uniqueSpanString := fmt.Sprintf("%s#%s#%s", execArn, stateName, stateEnteredTime)
+// extractTraceContextFromNestedStepFunctionContext extracts the root Step Function's execution ARN to generate the Trace ID
+func extractTraceContextFromNestedStepFunctionContext(event events.NestedStepFunctionPayload) (*TraceContext, error) {
+	if event.RootExecutionID == "" {
+		return nil, errorNoStepFunctionContextFound
+	}
+
+	tc, err := extractTraceContextFromStepFunctionContext(event.Payload)
```

Comment:
Being able to do this is why I liked the custom unmarshal route, otherwise we'd need to be dynamic about types to create some shared handler or parse out each of the values for each function here and in `extractTraceContextFromLambdaRootStepFunctionContext()`